impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -111,6 +111,9 @@ ContextVK::~ContextVK() {
     [[maybe_unused]] auto result = device_->waitIdle();
   }
   CommandPoolVK::ClearAllPools(this);
+  // Shut down the worker message loop before the context starts getting torn
+  // down.
```

Comment:
I'm not sure how to read this comment correctly.

The context is currently getting torn down right?

The extra reset here is guarding us against what? If someone adds another ivar after worker_message_loop_ that could somehow ref it?

Reply:
I've reworded the comment, let me know if that is more clear.

> The context is currently getting torn down right?

Right, but once we hit the closing bracket the ivars will start getting deleted in a undefined order (or at least a non-obvious order).  We want the threads all joined before we start doing that.

> The extra reset here is guarding us against what?

Guarding us from an unclear order of operations and adding more deterministic destruction.  This, for example, makes sure that the `shader_library_` is destructed on the raster thread instead of on some worker thread.